dpar-utils/src/util.rs

```diff
@@ -1,31 +1,63 @@
 use std::collections::HashMap;
-use std::fs::File;
+use std::fs::{self, File};
 use std::io::{BufRead, BufReader};
+use std::path::Path;
 
 use failure::Error;
 
-/// Read association strengths for dependency triples from a text file.
+/// Read association strengths for dependency triples from text files in
+/// a directory or from a single file.
 ///
-/// Such a text file consists of lines with the tab-separated format
+/// A text file consists of lines with the tab-separated format
 ///
 /// ~~~text,no_run
-/// [token+] [token+] [deprel+] association_strength
+/// [token+] [token+] association_strength
 /// ~~~
-pub fn associations_from_buf_read(
-    f: File,
+///
+/// The file is named with the dependency relation of which head-dependent
+/// pairs are listed with their association measure in the file.
+///
+/// E.g. the file is called `SUBJ.assoc` and contains all tokens related via the
+/// SUBJ relation and their association measure.
+pub fn associations_from_files(
+    dir: &Path,
 ) -> Result<HashMap<(String, String, String), f32>, Error> {
     let mut association_strengths: HashMap<(String, String, String), f32> = HashMap::new();
-    for l in BufReader::new(f).lines() {
-        let l = l.unwrap();
-        let line = l.split("\t").collect::<Vec<_>>();
-        association_strengths.insert(
-            (
-                line[0].to_string(),
-                line[1].to_string(),
-                line[2].to_string(),
-            ),
-            line[3].parse::<f32>().unwrap(),
-        );
+
+    if dir.is_dir() {
```

Comment:
I am not a fan of directory traversals:

* They are hard to get right (symlink cycles, etc.).
* People might give a directory with unrelated files or accidental temporary files, etc.

I think this can be done as a single file with the format:

```
delrel\tword1\tword2\tpmi
```

Reply:
Should we then just leave it the way it was? Because the previous version worked the way you describe it, except that the order of the columns was
```
word1\tword2\tdeprel\tpmi
```
and also the name was not correct in the sense that it should rather be `associations_from_file` than `associations_from_buf_read`.

Reply:
Yep. The order does not really matter.

Or call it `associations_from_buf_read` and make it take a `BufRead` ;).